### PR TITLE
Implement display product by category, remove product in cart, merge conflict

### PR DESCRIPTION
**Pull request by lelong199896**

### CODE REVIEW

File: src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java [hunk 1]
```diff
@@ -11,49 +11,77 @@
...
     public Product getProductById(int id) {
-        Product product = null;
+        Product selectProduct = null;
         try {
-            PreparedStatement preparedStatement = connection.prepareStatement(QueryConstant.get_product_by_id);
+            PreparedStatement preparedStatement = connection.prepareStatement(Query.get_product_by_id);
             preparedStatement.setInt(1, id);
             ResultSet resultSet = preparedStatement.executeQuery();
             while (resultSet.next()) {
+//                int userId = resultSet.getInt("user_id");
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java:58]: remove comment

File: src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java [hunk 2]
```diff
@@ -11,49 +11,77 @@
...
-                int userId = resultSet.getInt("user_id");
                 String productName = resultSet.getString("product_name");
-                double productQuantity = resultSet.getDouble("product_quantity");
-                double productPrice = resultSet.getDouble("product_cost");
-                boolean isDelete = resultSet.getBoolean("is_delete");
-                product = new Product(productId, userId, productName, productQuantity, productPrice, isDelete);
+                int productQuantity = resultSet.getInt("product_quantity");
+                double productCost = resultSet.getDouble("product_cost");
+//                int isActive = resultSet.getInt("is_delete");
+//                selectProduct = new Product(userId, productName, productQuantity, productCost ,isActive);
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java:64]: remove comments

**Review by Hgiang2 — APPROVED** [review]
ok

File: src/main/java/com/codegym/cglazadaplusproject/dao/UserDAO.java [hunk 3]
```diff
@@ -38,8 +38,12 @@ public List<User> getAllUser() {
     public User getUserById(int id) {
         User selectUser = null;
         try {
+<<<<<<< HEAD:src/main/java/com/codegym/cglazadaplusproject/dao/UserDAO.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/dao/UserDAO.java:41]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package com.codegym.cglazadaplusproject.model;
 
 public class Product {
+<<<<<<< HEAD:src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:4]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 5]
```diff
@@ -11,12 +12,25 @@ public class Product {
...
+    public Product(int userID, String productName, int productQuantity, double productCost, int isDelete) {
+        this.userID = userID;
+>>>>>>> b531d509370960c9c7f54efc27300b1dce3411be:cg-lazada-plus-project/src/main/java/com/codegym/cglazadaplusproject/model/Product.java
         this.productName = productName;
         this.productQuantity = productQuantity;
         this.productCost = productCost;
         this.isDelete = isDelete;
     }
 
+<<<<<<< HEAD:src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:33]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 6]
```diff
@@ -31,6 +45,44 @@ public int getProductUserId() {
...
+        this.productID = productID;
+    }
+
+    public int getUserID() {
+        return userID;
+    }
+
+    public void setUserID(int userID) {
+        this.userID = userID;
+>>>>>>> b531d509370960c9c7f54efc27300b1dce3411be:cg-lazada-plus-project/src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:85]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 7]
```diff
@@ -41,11 +93,19 @@ public void setProductName(String productName) {
         this.productName = productName;
     }
 
+<<<<<<< HEAD:src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:96]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 8]
```diff
@@ -41,11 +93,19 @@ public void setProductName(String productName) {
...
     }
 
     public void setProductQuantity(double productQuantity) {
+=======
+    public int getProductQuantity() {
+        return productQuantity;
+    }
+
+    public void setProductQuantity(int productQuantity) {
+>>>>>>> b531d509370960c9c7f54efc27300b1dce3411be:cg-lazada-plus-project/src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:108]: resolve this conflict code

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java [hunk 9]
```diff
@@ -57,6 +117,7 @@ public void setProductCost(double productCost) {
         this.productCost = productCost;
     }
 
+<<<<<<< HEAD:src/main/java/com/codegym/cglazadaplusproject/model/Product.java
```
Comment (tunguyendev0108) [inline src/main/java/com/codegym/cglazadaplusproject/model/Product.java:120]: resolve this conflict code

File: src/main/resources/db/db_11AM_13_12.sql [hunk 10]
```diff
@@ -124,9 +124,18 @@ foreign key (product_id)
 references product(product_id),
 foreign key (category_id)
 references category(category_id)
-);
-
-
+); 
+
+<<<<<<< HEAD:src/main/resources/db/db_11AM_13_12.sql
```
Comment (tunguyendev0108) [inline src/main/resources/db/db_11AM_13_12.sql:129]: resolve this conflict code

**Review by tunguyendev0108 — CHANGES_REQUESTED** [review]
resolve conflict code

File: src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java (hunk 1 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java:58]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java (hunk 2 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/dao/ProductDAO.java:64]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/dao/UserDAO.java (hunk 3 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/dao/UserDAO.java:41]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 4 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:4]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 5 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:33]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 6 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:85]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 7 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:96]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 8 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:108]: resolved

File: src/main/java/com/codegym/cglazadaplusproject/model/Product.java (hunk 9 above)
Comment (lelong199896) [reply src/main/java/com/codegym/cglazadaplusproject/model/Product.java:120]: resolved

File: src/main/resources/db/db_11AM_13_12.sql (hunk 10 above)
Comment (lelong199896) [reply src/main/resources/db/db_11AM_13_12.sql:129]: resolved

**Pull request merged**